test/emmy/examples/driven_pendulum_test.cljc

```diff
@@ -5,10 +5,19 @@
   (:require [clojure.test :refer [deftest is use-fixtures]]
             [emmy.env :as e :refer [up /]]
             [emmy.examples.driven-pendulum :as driven]
+            [emmy.expression.compile :refer [compile-state-fn*]]
             [emmy.simplify :refer [hermetic-simplify-fixture]]))
 
 (use-fixtures :each hermetic-simplify-fixture)
 
+(defn- gensym-fn
```

Comment:
you think it's worth putting this somewhere common? Folks are probably going to want something like this for the same reasons we want it in the tests, and I can't remember if we provide it already.

Reply:
Getting this far has made me want to crack the whip on this gensym business. Stay tuned.

(my current thought is something like this: there's no need to economize symbol generators. Each compile job can start out with a fresh one, that counts from 1, and can have various prefixes, so you'll get `["[y1, y2]" "[p3 p4]" " const _5 = ...] ` (modulo zero padding for the sort order, but we can get away with fewer digits since each function starts from scratch). If we thought 4 digits were enough, then they're short enough that the test expected outputs won't be horrible to look at, and the user will not need the ability to inject a generator because the compile outputs will always be stable).

Reply:
I have made it so that `monotonic-symbol-generator` is the only one left, but it's in the `analyze` namespace, which is not someplace we want our users to poke around. Open to suggestions as to where to move it (could put it in compile, for example)